src/Windows/Avalonia.Win32/DxgiSwapchain/HRESULT.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Avalonia.Win32.DxgiSwapchain
+{
+#nullable enable
+
+    public readonly unsafe partial struct HRESULT : IComparable, IComparable<HRESULT>, IEquatable<HRESULT>, IFormattable
```

Comment:
We already have HRESULT enum

Reply:
> We already have HRESULT enum

Would you be happy for me to add [FAILED](https://learn.microsoft.com/en-us/windows/win32/api/winerror/nf-winerror-failed) and [SUCCEEDED](https://learn.microsoft.com/en-us/windows/win32/api/winerror/nf-winerror-succeeded) properties to this enum to match the WinAPI C/C++ predefined macros?

Reply:
Removed, structs have been consolidated into a single cs file. Also turns out after using your above strategy for letting the runtime throw exceptions instead I no longer had any references to that HRESULT struct.

Reply:
Failed/Succeeded could be extension methods for the enum